maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java

```diff
@@ -305,25 +307,74 @@ public void promptForJavaHomeIfMissing(Server server) {
             String path = promptForMissingValueWithOptions(SDKConstants.OPENMRS_SDK_JDK_OPTION,
                     server.getJavaHome(), "path", paths, SDKConstants.OPENMRS_SDK_JDK_CUSTOM, null);
 
+            String requiredJdkVersion;
+            if (server.getPlatformVersion().charAt(0) == 1) {
+                requiredJdkVersion = "1.7";
+            }
+            else {
+                requiredJdkVersion = "1.8";
+            }
+
+            // Use default JAVA_HOME
             if (path.equals(paths.get(0))) {
-                // Use default JAVA_HOME
-                server.setJavaHome(null);
+                if (System.getProperty("java.version").startsWith(requiredJdkVersion)) {
+                    server.setJavaHome(null);
+                }
+                else if (System.getProperty("java.version").startsWith(requiredJdkVersion)) {
+                    server.setJavaHome(null);
+                }
+                else {
+                    showMessage("Your JAVA_HOME version doesn't fit platform requirements:");
+                    showMessage("JAVA_HOME version: " + System.getProperty("java.version"));
+                    showMessage("Required: " + requiredJdkVersion);
+                    promptForJavaHomeIfMissing(server);
+                }
             } else if (!isJavaHomeValid(path)) {
                 System.out.println(SDKConstants.OPENMRS_SDK_JDK_CUSTOM_INVALID);
                 promptForJavaHomeIfMissing(server);
             } else {
-                server.setJavaHome(path);
-                addJavaHomeToSdkProperties(path);
+                String jdkUnderSpecifiedPathVersion = extractJavaVersionFromPath(path);
+                if (jdkUnderSpecifiedPathVersion.startsWith(requiredJdkVersion)) {
+                    server.setJavaHome(path);
+                    addJavaHomeToSdkProperties(path);
+                }
+                else {
+                    showMessage("JDK in custom path (" + path + ") doesn't match platform requirements:");
+                    showMessage("JDK version: " + jdkUnderSpecifiedPathVersion);
+                    showMessage("Required: " + requiredJdkVersion);
+                    promptForJavaHomeIfMissing(server);
+                }
             }
         }
+    }
+
+    private String extractJavaVersionFromPath(String path) {
+        List<String> commands = new ArrayList<String>();
+        commands.add("java");
+        commands.add("-version");
+
+        ProcessBuilder processBuilder = new ProcessBuilder(commands);
+
+        processBuilder.directory(new File(path + File.separator + "bin"));
+
+        String result = null;
 
+        try {
+            final Process process = processBuilder.start();
+            BufferedReader stdInput = new BufferedReader(new InputStreamReader(process.getErrorStream()));
+
+            result = stdInput.readLine();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
 
+        return result.split("\"")[1];
```

Comment:
Could you please use some regex instead? It seems error prone if the ouput changes even slightly.